Server/mods/deathmatch/logic/CVehiclesConfig.cpp

```diff
@@ -0,0 +1,175 @@
+#include "CHandlingManager.h"
+
+CVehiclesConfig::CVehiclesConfig() : CXMLConfig(nullptr)
+{
+}
+
+bool CVehiclesConfig::Load()
+{
+    // Eventually destroy the previously loaded xml
+    if (m_pFile)
+    {
+        delete m_pFile;
+        m_pFile = nullptr;
+    }
+
+    // Load the XML
+    m_pFile = g_pServerInterface->GetXML()->CreateXML(GetFileName().c_str());
+    if (!m_pFile)
+    {
+        CLogger::ErrorPrintf("Error loading config file\n");
+        return false;
+    }
+
+    // Parse it
+    if (!m_pFile->Parse())
+    {
+        SString strParseErrorDesc;
+        m_pFile->GetLastError(strParseErrorDesc);
+        CLogger::ErrorPrintf("Error parsing %s - %s\n", *ExtractFilename(GetFileName()), *strParseErrorDesc);
+        return false;
+    }
+
+    // Grab the XML root node
+    CXMLNode* pRootNode = m_pFile->GetRootNode();
+    if (!pRootNode)
+    {
+        CLogger::ErrorPrintf("Missing root node ('config')\n");
+        return false;
+    }
+
+    RegisterVehicles(pRootNode);
+
+    return true;
+}
+
+bool CVehiclesConfig::Save()
+{
+    // Not implement
+    return false;
+}
+
+void CVehiclesConfig::RegisterVehicles(CXMLNode* pRootNode)
+{
+    for (auto it = pRootNode->ChildrenBegin(); it != pRootNode->ChildrenEnd(); ++it)
+    {
+        CXMLNode*       pNode = *it;
+        CXMLAttributes& pAttibites = pNode->GetAttributes();
+
+        CXMLAttribute*     pAttribute;
+        uint32_t           uiModelID;
+        SModelVehicleDefs* sModelData = new SModelVehicleDefs();
+        std::string_view   strType;
+        std::string_view   strVarianType;
+
+        pAttribute = pAttibites.Find("modelID");
+        if (pAttribute)
+            uiModelID = atoi(pAttribute->GetValue().c_str());
+        else
+            continue;
+
+        pAttribute = pAttibites.Find("name");
+        if (pAttribute)
+        {
+            std::string_view strName = pAttribute->GetValue();
+            const char*      pName = new char[strName.size()];
+            sModelData->strVehicleName = pName;
+        }
+
+        pAttribute = pAttibites.Find("type");
+        if (pAttribute)
+            strType = pAttribute->GetValue();
+
+        pAttribute = pAttibites.Find("variationType");
+        if (pAttribute)
+            strVarianType = pAttribute->GetValue();
+
+        pAttribute = pAttibites.Find("variantsCount");
+        if (pAttribute)
+            sModelData->uiVariantsCount = atoi(pAttribute->GetValue().c_str());
+
+        pAttribute = pAttibites.Find("attributes");
+        if (pAttribute)
+            sModelData->cAttributes = atoi(pAttribute->GetValue().c_str());
+
+        pAttribute = pAttibites.Find("maxPassngers");
+        if (pAttribute)
+            sModelData->uiMaxPassengers = atoi(pAttribute->GetValue().c_str());
+
+        pAttribute = pAttibites.Find("hasDoors");
+        if (pAttribute)
+            sModelData->bHasDoors = (pAttribute->GetValue().compare("true") == 0);
+
+        if (strType.compare("car") == 0)
```

Comment:
Would a switch...case be more efficient here?

Reply:
How?